ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -424,7 +424,11 @@ addBlock cfg blk m = Model {
       immutableChainHashes `isPrefixOf`
       map blockHash (Chain.toOldestFirst fork)
 
-    consideredCandidates = filter (extendsImmutableChain . fst) candidates
+    -- Considered candidates are candidates that extend the immutable part of
+    -- the chain. Furthermore, the current selected chain is not considered as a
+    -- candidate.
+    consideredCandidates = filter ((/= CPS.chainState (cps m)) . fst)
```

Comment:
With the other changes, is this now actually necessary to make any tests pass? Again, motivation is that this would be a difference to #284.

Reply:
If we use `consideredCandidates` for `valid'` as in https://github.com/input-output-hk/ouroboros-consensus/pull/279/files#r1291126148, that would also motivate not filtering out the current chain.